### ISSUE TITLE
0.10.12-beta-20181009 is case sensitive

### ISSUE THREAD

**Issue by CollinChaffin** [chocolatey/choco#1656]
### What You Are Seeing?

Catastrophic failures of both package installs as well as upgrades with latest choco beta v0.10.12-beta due to improper case sensitivity of package metadata IDs/nuget filename/nuget filepath.

### What is Expected?

Package names/IDs to my knowledge have never and should not be case-sensitive, so the second run result below is what I would expect for the outcome in the first run as well, and not a terminating error.

### How Did You Get This To Happen? (Steps to Reproduce)

````powershell
# EXECUTION RUN 1 - Prior installed package and lowercase was used successfully in the past and now fails with "not found" (FAILURE):

 C:\Scripts  
λ  cup --pre -y icofx -source \\TESTSERVER\ChocoPackages\icofx
Chocolatey v0.10.12-beta
Upgrading the following packages:
icofx
By upgrading you accept licenses for the packages.

icofx was not found with the source(s) listed.
 If you specified a particular version and are receiving this message, it is possible that the package name exists but the version does not.
 Version: ""; Source(s): "\\TESTSERVER\ChocoPackages\icofx"

Chocolatey upgraded 0/1 packages.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Warnings:
 - icofx - icofx was not found with the source(s) listed.



# EXECUTION RUN 2 - No changes to package data or path - as shown only change is the CASE of both package ID as well as package path (SUCCESSFUL):

 C:\Scripts  
λ  cup --pre -y IcoFx -source \\TESTSERVER\ChocoPackages\IcoFx
Chocolatey v0.10.12-beta
Upgrading the following packages:
IcoFx
By upgrading you accept licenses for the packages.
IcoFx v1.6.4.02 is the latest version available based on your source(s).

Chocolatey upgraded 0/1 packages.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).


 C:\Scripts  
λ

````
````
The case-specifics of the source folder:                              \\TESTSERVER\ChocoPackages\IcoFx
The case-specifics of the Nuget package file name:                    IcoFx.1.6.4.02.nupkg
The case-specifics of the metadata title/ID inside the package:       IcoFx
````

### Output Log

<details>
<summary>EXECUTION RUN 1 - FAILED</summary>

<p>

~~~sh
2018-10-11 15:55:33,761 2768 [DEBUG] - XmlConfiguration is now operational
2018-10-11 15:55:33,812 2768 [INFO ] - ============================================================
2018-10-11 15:55:34,040 2768 [INFO ] - Chocolatey v0.10.12-beta
2018-10-11 15:55:34,050 2768 [DEBUG] - Chocolatey is running on Windows v 6.1.7601.65536
2018-10-11 15:55:34,052 2768 [DEBUG] - Attempting to delete file "C:/ProgramData/chocolatey/choco.exe.old".
2018-10-11 15:55:34,054 2768 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\choco.exe.old".
2018-10-11 15:55:34,064 2768 [DEBUG] - Command line: "C:\ProgramData\chocolatey\choco.exe" upgrade --pre -y icofx -source \\TESTSERVER\ChocoPackages\icofx
2018-10-11 15:55:34,065 2768 [DEBUG] - Received arguments: upgrade --pre -y icofx -source \\TESTSERVER\ChocoPackages\icofx
2018-10-11 15:55:34,123 2768 [DEBUG] - RemovePendingPackagesTask is now ready and waiting for PreRunMessage.
2018-10-11 15:55:34,131 2768 [DEBUG] - Sending message 'PreRunMessage' out if there are subscribers...
2018-10-11 15:55:34,139 2768 [DEBUG] - [Pending] Removing all pending packages that should not be considered installed...
2018-10-11 15:55:35,440 2768 [DEBUG] - The source '\\TESTSERVER\ChocoPackages\icofx' evaluated to a 'normal' source type
2018-10-11 15:55:35,442 2768 [DEBUG] - 
NOTE: Hiding sensitive configuration data! Please double and triple 
 check to be sure no sensitive data is shown, especially if copying 
 output to a gist for review.
2018-10-11 15:55:35,452 2768 [DEBUG] - Configuration: CommandName='upgrade'|CacheLocation='C:\Temp\chocolatey'|
ContainsLegacyPackageInstalls='True'|
CommandExecutionTimeoutSeconds='2700'|WebRequestTimeoutSeconds='30'|
Sources='\\TESTSERVER\ChocoPackages\icofx'|SourceType='normal'|Debug='False'|
Verbose='False'|Trace='False'|Force='False'|Noop='False'|
HelpRequested='False'|UnsuccessfulParsing='False'|RegularOutput='True'|
QuietOutput='False'|PromptForConfirmation='False'|AcceptLicense='True'|
AllowUnofficialBuild='False'|Input='icofx'|AllVersions='False'|
SkipPackageInstallProvider='False'|PackageNames='icofx'|
Prerelease='True'|ForceX86='False'|OverrideArguments='False'|
NotSilent='False'|ApplyPackageParametersToDependencies='False'|
ApplyInstallArgumentsToDependencies='False'|IgnoreDependencies='False'|
AllowMultipleVersions='False'|AllowDowngrade='False'|
ForceDependencies='False'|Information.PlatformType='Windows'|
Information.PlatformVersion='6.1.7601.65536'|
Information.PlatformName='Windows 7'|
Information.ChocolateyVersion='0.10.12.0'|
Information.ChocolateyProductVersion='0.10.12-beta'|
Information.FullName='choco, Version=0.10.12.0, Culture=neutral, PublicKeyToken=79d02ea9cad655eb'|

Information.Is64BitOperatingSystem='True'|
Information.Is64BitProcess='True'|Information.IsInteractive='True'|
Information.UserName='Admin1'|Information.UserDomainName='ONYX'|
Information.IsUserAdministrator='True'|
Information.IsUserSystemAccount='False'|
Information.IsUserRemoteDesktop='False'|
Information.IsUserRemote='False'|Information.IsProcessElevated='True'|
Information.IsLicensedVersion='False'|Information.LicenseType='Foss'|
Features.AutoUninstaller='False'|Features.ChecksumFiles='False'|
Features.AllowEmptyChecksums='True'|
Features.AllowEmptyChecksumsSecure='True'|
Features.FailOnAutoUninstaller='False'|
Features.FailOnStandardError='False'|Features.UsePowerShellHost='True'|
Features.LogEnvironmentValues='False'|Features.LogWithoutColor='False'|
Features.VirusCheck='False'|
Features.FailOnInvalidOrMissingLicense='False'|
Features.IgnoreInvalidOptionsSwitches='True'|
Features.UsePackageExitCodes='True'|
Features.UseFipsCompliantChecksums='False'|
Features.ShowNonElevatedWarnings='True'|
Features.ShowDownloadProgress='True'|
Features.StopOnFirstPackageFailure='False'|
Features.UseRememberedArgumentsForUpgrades='True'|
Features.IgnoreUnfoundPackagesOnUpgradeOutdated='False'|
Features.RemovePackageInformationOnUninstall='True'|
Features.ScriptsCheckLastExitCode='False'|
ListCommand.LocalOnly='False'|
ListCommand.IdOnly='False'|ListCommand.IncludeRegistryPrograms='False'|
ListCommand.PageSize='25'|ListCommand.Exact='False'|
ListCommand.ByIdOnly='False'|ListCommand.ByTagOnly='False'|
ListCommand.IdStartsWith='False'|ListCommand.OrderByPopularity='False'|
ListCommand.ApprovedOnly='False'|
ListCommand.DownloadCacheAvailable='False'|
ListCommand.NotBroken='False'|
ListCommand.IncludeVersionOverrides='False'|
UpgradeCommand.FailOnUnfound='False'|
UpgradeCommand.FailOnNotInstalled='False'|
UpgradeCommand.NotifyOnlyAvailableUpgrades='False'|
UpgradeCommand.ExcludePrerelease='False'|
NewCommand.AutomaticPackage='False'|
NewCommand.UseOriginalTemplate='False'|SourceCommand.Command='unknown'|
SourceCommand.Priority='0'|SourceCommand.BypassProxy='False'|
SourceCommand.AllowSelfService='False'|
SourceCommand.VisibleToAdminsOnly='False'|
FeatureCommand.Command='unknown'|ConfigCommand.Command='unknown'|
PinCommand.Command='unknown'|OutdatedCommand.IgnorePinned='False'|
Proxy.BypassList='192.168.99.100'|Proxy.BypassOnLocal='True'|
2018-10-11 15:55:35,454 2768 [DEBUG] - _ Chocolatey:ChocolateyUpgradeCommand - Normal Run Mode _
2018-10-11 15:55:35,458 2768 [INFO ] - Upgrading the following packages:
2018-10-11 15:55:35,460 2768 [INFO ] - icofx
2018-10-11 15:55:35,462 2768 [INFO ] - By upgrading you accept licenses for the packages.
2018-10-11 15:55:35,806 2768 [WARN ] - 
icofx was not found with the source(s) listed.
 If you specified a particular version and are receiving this message, it is possible that the package name exists but the version does not.
 Version: ""; Source(s): "\\TESTSERVER\ChocoPackages\icofx"
2018-10-11 15:55:35,819 2768 [WARN ] - 
Chocolatey upgraded 0/1 packages. 
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
2018-10-11 15:55:35,821 2768 [INFO ] - 
2018-10-11 15:55:35,823 2768 [WARN ] - Warnings:
2018-10-11 15:55:35,824 2768 [WARN ] -  - icofx - icofx was not found with the source(s) listed.
2018-10-11 15:55:35,830 2768 [DEBUG] - Sending message 'PostRunMessage' out if there are subscribers...
2018-10-11 15:55:35,835 2768 [DEBUG] - Exiting with 0
~~~

</p>

</details>

<details>
<summary>EXECUTION RUN 2 - SUCCESSFUL - ONLY CHANGE IS CASE SENSITIVITY</summary>

<p>

~~~sh
2018-10-11 16:02:09,600 15988 [DEBUG] - XmlConfiguration is now operational
2018-10-11 16:02:09,652 15988 [INFO ] - ============================================================
2018-10-11 16:02:09,883 15988 [INFO ] - Chocolatey v0.10.12-beta
2018-10-11 16:02:09,898 15988 [DEBUG] - Chocolatey is running on Windows v 6.1.7601.65536
2018-10-11 16:02:09,901 15988 [DEBUG] - Attempting to delete file "C:/ProgramData/chocolatey/choco.exe.old".
2018-10-11 16:02:09,903 15988 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\choco.exe.old".
2018-10-11 16:02:09,913 15988 [DEBUG] - Command line: "C:\ProgramData\chocolatey\choco.exe" upgrade --pre -y IcoFx -source \\TESTSERVER\ChocoPackages\IcoFx
2018-10-11 16:02:09,914 15988 [DEBUG] - Received arguments: upgrade --pre -y IcoFx -source \\TESTSERVER\ChocoPackages\IcoFx
2018-10-11 16:02:09,972 15988 [DEBUG] - RemovePendingPackagesTask is now ready and waiting for PreRunMessage.
2018-10-11 16:02:09,980 15988 [DEBUG] - Sending message 'PreRunMessage' out if there are subscribers...
2018-10-11 16:02:09,989 15988 [DEBUG] - [Pending] Removing all pending packages that should not be considered installed...
2018-10-11 16:02:11,302 15988 [DEBUG] - The source '\\TESTSERVER\ChocoPackages\IcoFx' evaluated to a 'normal' source type
2018-10-11 16:02:11,304 15988 [DEBUG] - 
NOTE: Hiding sensitive configuration data! Please double and triple 
 check to be sure no sensitive data is shown, especially if copying 
 output to a gist for review.
2018-10-11 16:02:11,314 15988 [DEBUG] - Configuration: CommandName='upgrade'|CacheLocation='C:\Temp\chocolatey'|
ContainsLegacyPackageInstalls='True'|
CommandExecutionTimeoutSeconds='2700'|WebRequestTimeoutSeconds='30'|
Sources='\\TESTSERVER\ChocoPackages\IcoFx'|SourceType='normal'|Debug='False'|
Verbose='False'|Trace='False'|Force='False'|Noop='False'|
HelpRequested='False'|UnsuccessfulParsing='False'|RegularOutput='True'|
QuietOutput='False'|PromptForConfirmation='False'|AcceptLicense='True'|
AllowUnofficialBuild='False'|Input='IcoFx'|AllVersions='False'|
SkipPackageInstallProvider='False'|PackageNames='IcoFx'|
Prerelease='True'|ForceX86='False'|OverrideArguments='False'|
NotSilent='False'|ApplyPackageParametersToDependencies='False'|
ApplyInstallArgumentsToDependencies='False'|IgnoreDependencies='False'|
AllowMultipleVersions='False'|AllowDowngrade='False'|
ForceDependencies='False'|Information.PlatformType='Windows'|
Information.PlatformVersion='6.1.7601.65536'|
Information.PlatformName='Windows 7'|
Information.ChocolateyVersion='0.10.12.0'|
Information.ChocolateyProductVersion='0.10.12-beta'|
Information.FullName='choco, Version=0.10.12.0, Culture=neutral, PublicKeyToken=79d02ea9cad655eb'|

Information.Is64BitOperatingSystem='True'|
Information.Is64BitProcess='True'|Information.IsInteractive='True'|
Information.UserName='Admin1'|Information.UserDomainName='ONYX'|
Information.IsUserAdministrator='True'|
Information.IsUserSystemAccount='False'|
Information.IsUserRemoteDesktop='False'|
Information.IsUserRemote='False'|Information.IsProcessElevated='True'|
Information.IsLicensedVersion='False'|Information.LicenseType='Foss'|
Features.AutoUninstaller='False'|Features.ChecksumFiles='False'|
Features.AllowEmptyChecksums='True'|
Features.AllowEmptyChecksumsSecure='True'|
Features.FailOnAutoUninstaller='False'|
Features.FailOnStandardError='False'|Features.UsePowerShellHost='True'|
Features.LogEnvironmentValues='False'|Features.LogWithoutColor='False'|
Features.VirusCheck='False'|
Features.FailOnInvalidOrMissingLicense='False'|
Features.IgnoreInvalidOptionsSwitches='True'|
Features.UsePackageExitCodes='True'|
Features.UseFipsCompliantChecksums='False'|
Features.ShowNonElevatedWarnings='True'|
Features.ShowDownloadProgress='True'|
Features.StopOnFirstPackageFailure='False'|
Features.UseRememberedArgumentsForUpgrades='True'|
Features.IgnoreUnfoundPackagesOnUpgradeOutdated='False'|
Features.RemovePackageInformationOnUninstall='True'|
Features.ScriptsCheckLastExitCode='False'|
ListCommand.LocalOnly='False'|
ListCommand.IdOnly='False'|ListCommand.IncludeRegistryPrograms='False'|
ListCommand.PageSize='25'|ListCommand.Exact='False'|
ListCommand.ByIdOnly='False'|ListCommand.ByTagOnly='False'|
ListCommand.IdStartsWith='False'|ListCommand.OrderByPopularity='False'|
ListCommand.ApprovedOnly='False'|
ListCommand.DownloadCacheAvailable='False'|
ListCommand.NotBroken='False'|
ListCommand.IncludeVersionOverrides='False'|
UpgradeCommand.FailOnUnfound='False'|
UpgradeCommand.FailOnNotInstalled='False'|
UpgradeCommand.NotifyOnlyAvailableUpgrades='False'|
UpgradeCommand.ExcludePrerelease='False'|
NewCommand.AutomaticPackage='False'|
NewCommand.UseOriginalTemplate='False'|SourceCommand.Command='unknown'|
SourceCommand.Priority='0'|SourceCommand.BypassProxy='False'|
SourceCommand.AllowSelfService='False'|
SourceCommand.VisibleToAdminsOnly='False'|
FeatureCommand.Command='unknown'|ConfigCommand.Command='unknown'|
PinCommand.Command='unknown'|OutdatedCommand.IgnorePinned='False'|
Proxy.BypassList='192.168.99.100'|Proxy.BypassOnLocal='True'|
2018-10-11 16:02:11,316 15988 [DEBUG] - _ Chocolatey:ChocolateyUpgradeCommand - Normal Run Mode _
2018-10-11 16:02:11,320 15988 [INFO ] - Upgrading the following packages:
2018-10-11 16:02:11,322 15988 [INFO ] - IcoFx
2018-10-11 16:02:11,324 15988 [INFO ] - By upgrading you accept licenses for the packages.
2018-10-11 16:02:11,656 15988 [INFO ] - IcoFx v1.6.4.02 is the latest version available based on your source(s).
2018-10-11 16:02:11,670 15988 [WARN ] - 
Chocolatey upgraded 0/1 packages. 
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
2018-10-11 16:02:11,674 15988 [DEBUG] - Sending message 'PostRunMessage' out if there are subscribers...
2018-10-11 16:02:11,680 15988 [DEBUG] - Exiting with 0
~~~

</p>

</details>

**Comment by ferventcoder** [comment]
@CollinChaffin this is correct, there should not be case sensitivity.

**Comment by ferventcoder** [comment]
Do keep in mind there is a reason this is a pre-release. It is to drive these kinds of issues out before the full release.

**Comment by ferventcoder** [comment]
For now, jump back to the stable 0.10.11 release. Appreciate the comprehensive logs!

**Comment by ferventcoder** [comment]
Found the issue - The changes in #1397 caused a case sensitivity issue.

**Comment by CollinChaffin** [comment]
Perfect, thanks Rob!  As usual I don't run the beta on anything mission critical and hesitantly used the word "catastrophic" really just because of the severity of the failure itself. :)

Glad you found the issue I can confirm that I've already pulled down beta 20181011 and tested and can confirm the issue resolved!  Good work!

**Comment by ferventcoder** [comment]
This has been fixed with https://github.com/chocolatey/choco/commit/742cf64d7cb21b6c0755e662d777b2e3df4b26ae and will be released as part of 0.10.12.
